### PR TITLE
fix: delete payments when a publisher allowance is deleted

### LINKED ISSUE #186: Delete payments when a publisher allowance is paid
### Describe the bug

Deleting a publisher (i.e. nakaphoto) does not delete related payments.
Currently we only delete the `allowance` entry not the payments.

### Screenshots

- Delete the nakaphoto entry  
    
- Payment still visible   
    

### To Reproduce

1. Visit 'Websites'
1. Delete an entry which has payments
1. Open the popup
1. Related (outgoing) transactions are still visible

### Expected behavior

After deleting a publisher the related payments should be gone as well.

### Additional context

- Allowance delete is happening here:  
    https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/actions/allowances/delete.ts
- In case of getting an allowance we're adding related payments here:
    https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/actions/allowances/getById.ts#L32-L36
- Getting all related payments and delete them:
    https://dexie.org/docs/Collection/Collection.delete()

This issue is related to #1505

### LINKED ISSUE #1505: Improve alert-text when deleting a website
### Describe the bug

If I go to allowance and click on delete, appear this message: Are you sure you want to delete this website?. But I think the text can be improved to something like: If you delete this website, you will loose all history of transactions you did on it, like total of payments and sats spent. Are you sure you want to delete this website?

### Screenshots (If Any)

Captura de tela de 2022-09-22 10-53-50

### To Reproduce

See the screenshot

### Expected behavior

I think the text can be improved to something like: If you delete this website, you will loose all history of transactions you did on it, like total of payments and sats spent. Are you sure you want to delete this website?

### Alby information

- Alby Version: 1.15.1

### Are you working on this issue?

No

### PR DESCRIPTION
**Pull request by qqqzhch**
### Describe the changes you have made in this PR

-  After deleting a publisher the related payments should be gone as wel
-   The text can be improved to something like: If you delete this website, you will loose all history of transactions you did on it, like total of payments and sats spent. Are you sure you want to delete this website?

_A clear and concise description of what you want to happen_

### Link this PR to an issue [optional]

- Fixes #186
- Fixes #1505

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Added unit tests related to website deletion

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/i18n/locales/es/translation.json [hunk 1]
```diff
@@ -546,7 +546,7 @@
   },
   "components": {
     "allowance_menu": {
-      "confirm_delete": "¿Está seguro de que desea eliminar este sitio web?",
+      "confirm_delete": "¿Está seguro de que desea eliminar este sitio web? \nSi elimina este sitio web, perderá todos los registros históricos de transacciones que realiza en él, como el total de pagos y el SAT gastado.",
```
Comment (escapedcat) [inline src/i18n/locales/es/translation.json:549]: Thanks, but no need to translate others than English. People are using [Weblate](https://hosted.weblate.org/projects/getalby-lightning-browser-extension/getalby-lightning-browser-extension/) to add missign translations and also can watch projects I think.

File: src/extension/background-script/actions/allowances/__tests__/delete.test.ts [hunk 2]
```diff
@@ -68,6 +74,15 @@ describe("delete allowance", () => {
       .reverse()
       .sortBy("lastPaymentAt");
 
+    if (dbAllowance) {
+      paymentsCount = await db.payments
+        .where("host")
+        .equalsIgnoreCase(dbAllowance.host)
+        .count();
+
+      expect(paymentsCount).toEqual(0);
```
Comment (escapedcat) [inline src/extension/background-script/actions/allowances/__tests__/delete.test.ts:83]: Does it make sense to first check if payments exists i.e. 
```ts
expect(paymentsCount).toEqual(1);
```
and then check for `0`?

**Review by escapedcat — COMMENTED** [review]
Nice thanks

File: src/extension/background-script/actions/allowances/__tests__/delete.test.ts (hunk 2 above)
Comment (qqqzhch) [reply src/extension/background-script/actions/allowances/__tests__/delete.test.ts:83]: Yeah, that's what I thought

**Comment by escapedcat** [comment]
apart from my comments: ack

File: src/extension/background-script/actions/allowances/__tests__/delete.test.ts (hunk 2 above)
Comment (escapedcat) [reply src/extension/background-script/actions/allowances/__tests__/delete.test.ts:83]: Cool, wanna add it?

File: src/i18n/locales/es/translation.json (hunk 1 above)
Comment (qqqzhch) [reply src/i18n/locales/es/translation.json:549]: Well, I saw this website during the translation test. Keep the new text in English and keep the original text in other languages，

**Review by escapedcat — APPROVED** [review]

**Comment by reneaaron** [comment]
From a user perspective that makes no sense to me . I just wanted to delete my "budget" for this site. What's the argument for deleting payment data along with an allowance?

**Comment by bumi** [comment]
If you want to delete the budget you should not delete the website - this "disconnects" the website and next time you have to enable it again.

**Comment by bumi** [comment]
I think the consensus was now that we want to keep the payments and only the allowance should be deleted. 
Otherwise people are not aware that they loose their payments.

**Pull request closed without merging**